### ISSUE TITLE
Configure error when using  Sundials 6.1.1

### ISSUE THREAD

**Issue by sagitter** [boutproject/BOUT-dev#2501]
Hi all.

`configure` is failing when i try to compile `bout++-4.4.1` against  `sundials-6.1.1` (GCC 12.0.1 20220212):

```
configure:11945: Searching for SUNDIALS IDA library
configure:12079: Checking for IDA header files
configure:12084: checking for /usr/include/mpich-x86_64/ida/ida.h
configure:12099: result: yes
configure:12113: checking for /usr/include/mpich-x86_64/ida/ida_spgmr.h
configure:12128: result: no
configure:12142: checking for /usr/include/mpich-x86_64/ida/ida_bbdpre.h
configure:12157: result: yes
configure:12171: checking for /usr/include/mpich-x86_64/nvector/nvector_parallel.h
configure:12186: result: yes
configure:12200: checking for /usr/include/mpich-x86_64/sundials/sundials_types.h
configure:12215: result: yes
configure:12390: Found IDA include path: /usr/include/mpich-x86_64
configure:12409: checking if SUNDIALS IDA library path is /usr/lib64/mpich/lib
configure:12431: mpic++ -o conftest -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -DHAS_PRETTY_FUNCTION -Wextra -Wnull-dereference -Wno-cast-function-type -DCHECK=2 -DSIGHANDLE -DLOGCOLOR -DBOUT_VERSION_STRING="\"4.4.1\"" -DBOUT_VERSION_DOUBLE=4.41  -DBACKTRACE -fPIC -DBOUT_HAS_FFTW -DNCDF4 -DHDF5 -DBOUT_HAS_PETSC  -I/usr/include/mpich-x86_64 -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -L/usr/lib64/mpich/lib conftest.cpp -lm   -lsundials_ida -lsundials_nvecparallel  -lsundials_arkode -lsundials_nvecparallel -lsundials_ida -L/usr/lib64 -lflexiblas >&5
conftest.cpp: In function 'int main()':
conftest.cpp:48:10: error: too few arguments to function 'void* IDACreate(SUNContext)'
   48 | IDACreate();
      | ~~~~~~~~~^~
In file included from conftest.cpp:41:
/usr/include/mpich-x86_64/ida/ida.h:107:23: note: declared here
  107 | SUNDIALS_EXPORT void *IDACreate(SUNContext sunctx);
      |                       ^~~~~~~~~
configure:12431: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "BOUT++"
| #define PACKAGE_TARNAME "bout--"
| #define PACKAGE_VERSION "4.4.1"
| #define PACKAGE_STRING "BOUT++ 4.4.1"
| #define PACKAGE_BUGREPORT "bd512@york.ac.uk"
| #define PACKAGE_URL ""
| #define HAVE_MPI 1
| #define HAVE_CXX11 1
| #define HAVE_LIBM 1
| #define HAVE_STDIO_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_UNISTD_H 1
| #define STDC_HEADERS 1
| #define HAVE_MALLOC_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_MALLOC 1
| #define HAVE_REALLOC 1
| #define HAVE_VPRINTF 1
| #define HAVE_POPEN 1
| #define HAVE_BACKTRACE 1
| #define HAVE_EXECINFO_H 1
| #define HAVE_DLFCN_H 1
| #define HAVE_HDF5 1
| #define HAVE__USR_INCLUDE_MPICH_X86_64_IDA_IDA_H 1
| #define HAVE__USR_INCLUDE_MPICH_X86_64_IDA_IDA_BBDPRE_H 1
| #define HAVE__USR_INCLUDE_MPICH_X86_64_NVECTOR_NVECTOR_PARALLEL_H 1
| #define HAVE__USR_INCLUDE_MPICH_X86_64_SUNDIALS_SUNDIALS_TYPES_H 1
| /* end confdefs.h.  */
| 
| 
|     #include <nvector/nvector_parallel.h>
|     #include <ida/ida.h>
|     extern void foo(N_Vector);
| 
| 
| int
| main (void)
| {
| IDACreate();
|   ;
|   return 0;
| }
configure:12439: result: no
configure:12462: error: in `/builddir/build/BUILD/BOUT++-v4.4.1/build_mpich':
configure:12464: error: Cannot compile IDA program
See `config.log' for more details
```

[config.log](https://github.com/boutproject/BOUT-dev/files/8074138/config.log) is attached

**Comment by ZedThree** [comment]
It looks like the interface has changed a little in 6.1.1, and we need to create and manage a `SUNContext` object.

As it stands, this means we currently only support SUNDIALS < 6.0.0. Please try with SUNDIALS 5.7.0

**Comment by dschwoerer** [comment]
This appears to be part of the effort to get the most recent SUNDIALS into fedora - so an older version defeats the purpose ;-)

According to [the release notes](https://computing.llnl.gov/projects/sundials/release-history) there is a script to do the changes:
` > ./upgrade-to-sundials-6-from-5.sh <files to update>`
The script can be found [here](https://github.com/LLNL/sundials/releases/download/v6.0.0/upgrade-to-sundials-6-from-5.sh) but I didn't get to testing it yet ...

**Comment by ZedThree** [comment]
Fixed in `next`, will be in BOUT++ 5.0
